src/snabbdom.ts

```diff
@@ -57,10 +57,27 @@ export function init(modules: Array<Partial<Module>>, domApi?: DOMAPI) {
     }
   }
 
-  function emptyNodeAt(elm: Element) {
+  function toVNode(elm: Element): VNode {
     const id = elm.id ? '#' + elm.id : '';
-    const c = elm.className ? '.' + elm.className.split(' ').join('.') : '';
-    return vnode(api.tagName(elm).toLowerCase() + id + c, {}, [], undefined, elm);
+    const cn = elm.getAttribute('class');
+    const c = cn ? '.' + cn.split(' ').join('.') : '';
+    const sel = api.tagName(elm).toLowerCase() + id + c;
+    const attrs: any = {};
+    for (let i = 0, elmAttrs = elm.attributes, n = elmAttrs.length; i < n; i++) {
+      attrs[elmAttrs[i].nodeName] = elmAttrs[i].nodeValue;
+    }
+    delete attrs.id;
+    delete attrs.class;
+    const children: Array<VNode | string> = [];
+    for (let i = 0, elmChildren = elm.childNodes, n = elmChildren.length; i < n; i++) {
+      const node = elmChildren[i];
+      if (node.nodeType === 1) {
+        children.push(toVNode(elmChildren[i] as Element));
+      } else if (node.nodeType === 2) {
```

Comment:
Did you test this? `Text` nodes should have a `nodeType` of `3`.
https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType#Node_type_constants

Also, I'd consider creating a `nodeType` function in `htmldomapi.ts` instead, to be consistent with how the `tagName` is accessed. Maybe also declare the `1` and `3` as constants there?